src/main/scala/tscfg/generators/java/JavaGen.scala

```diff
@@ -144,16 +151,34 @@ class JavaGen(genOpts: GenOpts) extends Generator(genOpts) {
       ""
     )
 
-    val classStr = {
-      s"""public ${if (isRoot) "" else "static "}class $classNameAdjusted {
-         |  $classDeclMembersStr$classMemberGettersStr
-         |  $membersStr
-         |  public $classNameAdjusted($ctorParams) {
-         |    $errHandlingDecl$ctorMembersStr$errHandlingDispatch
-         |  }$elemAccessorsStr
-         |$rootAuxClasses}
-         |""".stripMargin
-    }
+    val extendsString = abstractClassName.map("extends " + _).getOrElse("")
+    val superString = abstractClassName.map(_ => "super(c, parentPath, $tsCfgValidator);").getOrElse("") // if parent class name is defined a super call is needed
+
+    val classStr =
+      ot match {
+        case _: ObjectType =>
+          s"""public ${
+            if (isRoot) "" else "static "
+          }class $classNameAdjusted $extendsString {
+             |  $classDeclMembersStr$classMemberGettersStr
+             |  $membersStr
+             |  public $classNameAdjusted($ctorParams) {
```

Comment:
(2b) and use it like this to avoid the empty line after the `{` that is being added to all other cases where there's no `super` involved:
```scala
  public $classNameAdjusted($ctorParams) {$superString
```

Reply:
@johanneshiry Those observations (1) and (2) above are to avoid changing the java generation wrt to previous versions of tscfg for config specs not involving the new "abstract" feature.  I've seen user also committing the generated classes in version control and these adjustments, no only helps with a little better formatted code, but also avoids such users deal with unnecessary changes if running the new version of the generator.